src/main/java/edu/harvard/iq/dataverse/DataverseServiceBean.java

```diff
@@ -195,6 +196,19 @@ public Dataverse findRootDataverse() {
         return em.createNamedQuery("Dataverse.findRoot", Dataverse.class).getSingleResult();
     }
     
+    
+    //Similarly - if the above throws that exception, do we need to catch it here?
+    //ToDo - consider caching?
+    public String getRootDataverseName() {
+        Dataverse root = findRootDataverse();
+        if(root==null) {
+            logger.warning("Failed to find root dataverse!");
+            return "";
```

Comment:
if no root dataverse, should we throw an exception instead? this seems like something that won't happen, but if it does, should be ignored.

Reply:
Hmm - Or is it needed at all? - findRootDataverse should throw an exception if it isn't found - can it return null?

Reply:
Yeah, that's what I meant; let it throw the exception. (it will throw a NoResultException)

Reply:
"null is not an option" (TM)